Mathlib/Data/Set/Card.lean

```diff
@@ -1055,5 +1056,132 @@ theorem ncard_eq_three : s.ncard = 3 ↔ ∃ x y z, x ≠ y ∧ x ≠ z ∧ y 
+    have hynu : y ∉ u := fun hys ↦ by simpa [hst.1] using (subset_union_right hys : y ∈ t ∪ u)
+    have hxnt : x ∉ t := fun hxs ↦ by simpa [hst.1] using (subset_union_left hxs : x ∈ t ∪ u)
+    have htuf : (t ∪ u).Finite := by
+      rw [hst.1]
+      exact Finite.diff hs {x, y}
+    constructor <;> simp [hxnu, hynt, hst.2.1, hxy.2.2.symm,
+      ncard_insert_of_not_mem hxnt (finite_union.mp htuf).1,
+      ncard_insert_of_not_mem hynu (finite_union.mp htuf).2, hst.2.2]
+termination_by s.ncard
+decreasing_by
+· simp_wf
+  refine ncard_lt_ncard ?_ hs
+  exact ⟨diff_subset, by
+    rw [not_subset_iff_exists_mem_not_mem]
+    use x
+    exact ⟨hxy.1, by simp only [mem_diff, mem_insert_iff, mem_singleton_iff, true_or,
+      not_true_eq_false, and_false, not_false_eq_true]⟩⟩
+
+theorem exists_union_disjoint_cardinal_eq_of_infinite [DecidableEq α] (s : Set α)
+    (h : s.Infinite) : ∃ (t u : Set α),
+    t ∪ u = s ∧ Disjoint t u ∧ Cardinal.mk t = Cardinal.mk u := by
+  have f : s ⊕ s ≃ s := by
+      have : Inhabited (s ⊕ s ≃ s) := by
+        apply Classical.inhabited_of_nonempty
+        rw [← Cardinal.eq, Cardinal.mk_sum, Cardinal.add_eq_max (by
+          rw [@Cardinal.aleph0_le_lift]
+          exact Cardinal.infinite_iff.mp (infinite_coe_iff.mpr h)
+          )]
+        simp only [Cardinal.lift_id, max_self]
+      exact this.default
+  use Subtype.val '' (f '' (Sum.inl '' univ)), Subtype.val '' (f '' (Sum.inr '' univ))
+  constructor
+  · ext v
+    simp only [image_univ, mem_union, mem_image, mem_image_equiv, mem_range,
+      Subtype.exists, exists_and_right, exists_eq_right]
+    refine ⟨fun h ↦ by
+      cases' h with hl hr
+      · obtain ⟨hvu, _⟩ := hl
+        exact hvu
+      · obtain ⟨hvu, _⟩ := hr
+        exact hvu, ?_⟩
+    · intro hv
+      rw [← exists_or]
+      use hv
+      simp only [Sum.exists, Sum.inl.injEq, exists_subtype_mk_eq_iff, exists_eq, true_and,
+        Subtype.exists, reduceCtorEq, exists_false, false_and, or_false, Sum.inr.injEq, false_or]
+      obtain ⟨a, ha⟩ := f.surjective ⟨v, hv⟩
+      rw [← ha]
+      simp only [EmbeddingLike.apply_eq_iff_eq]
+      cases' a with l r
+      · left; use l, l.coe_prop
+      · right; use r, r.coe_prop
+  · constructor
+    · simp only [image_univ, ← image_comp]
+      apply disjoint_image_of_injective (by
+        simp only [Subtype.val_injective, Function.Injective.of_comp_iff, f.injective])
+      rw [disjoint_right]
+      intro a ha
+      simp only [mem_range, Subtype.exists, not_exists] at *
+      intro v hv
+      obtain ⟨_, _, h⟩ := ha
+      rw [← h]
+      exact Sum.inl_ne_inr
+    · simp only [image_univ, Subtype.val_injective, Cardinal.mk_image_eq, f.injective,
+        Sum.inl_injective, Cardinal.mk_range_eq, Sum.inr_injective]
+
+
+theorem exists_union_disjoint_cardinal_eq_iff [DecidableEq α] (s : Set α) :
+    Even (s.ncard) ↔ ∃ (t u : Set α), t ∪ u = s ∧ Disjoint t u ∧ Cardinal.mk t = Cardinal.mk u := by
```

Comment:
Can you use `#` notation for `Cardinal.mk` here?

Reply:
I might be missing something, but I could only find `#` notation for Finset, does it also exist for Set?